gcp_variant_transforms/vcf_to_bq.py

```diff
@@ -218,44 +219,92 @@ def run(argv=None):
       known_args.minimal_vep_alt_matching,
       counter_factory)
 
-  partitioner = variant_partition.VariantPartition()
+  if known_args.partition_config_path:
```

Comment:
I've been thinking about this and I think perhaps a combination of your original suggestion (about keeping a single variable named `variants` and passing that around) and some cleanup can be more readable. Something like:
```
create partitioner (can be None if not needed)
read VCF files and filter
if partitioner:
  num_partitions = partitioner.get_num_partitions()
  variants |= 'PartitionVariants' >> beam.Partition(
        partition_variants.PartitionVariants(partitioner), num_partitions)
  # does 'del variants[partitioner.get_residual_partition_index()]' work? if not we need a way to remove it from here.
  # perhaps by reconstructing another array that doesn't have that particular index?
else:
  num_partitions = 1
  variants = [variants]

for i in range(num_partitions):
  variants[i] |= merge
  variants[i] |= convert to processed_variant # add anything else that may come along later

if partitioner.should_flatten():  # based on whether there was a config file
  variants = [variants | 'FlattenPartitions' >> beam.Flatten()]
  num_partitions = 1

for i in range(num_partitions):
  output to BigQuery using table suffix based on i
```
We'd need to think about the merge step when/if we implement gVCF merging as we'd need to take care of regions that span multiple partitions (if not done by chromosome), but we can figure that out later.
Lemme know what you think!

Reply:
I implemented this and code now looks much nicer, please take a look and let me know what you think.
Meanwhile I have two test runs and I will add the integration tests.